cmd/genbindings/emitcabi.go

```diff
@@ -1262,6 +1347,60 @@ extern "C" {
 
 		}
 
+		if len(virtualMethods) > 0 {
+			// This is a subclassed class. In that case, we allow calling
+			// protected methods
+			// This is a standalone function, but it can access the protected
+			// method via a friend declaration
+
+			for _, m := range protectedMethods {
+
+				vbpreamble, vbforwarding := emitParametersCABI2CppForwarding(m.Parameters, "\t\t")
+				vbCallTarget := "self_cast->" + m.CppCallTarget() + "(" + vbforwarding + ")"
+
+				assignStmts := emitAssignCppToCabi("\treturn ", m.ReturnType, vbCallTarget)
+
+				// FIXME(hack): In some platforms, instantiating a protected
+				// enum fails in friend context:
+				//
+				//     QAbstractItemView::State _ret = self_cast->state();
+				//     error: 'State' is a protected member of 'QAbstractItemView'
+				//
+				// @ref https://stackoverflow.com/q/52191903
+				// However, it works fine on most other platforms. Probably this
+				// is a GCC vs Clang difference.
+				//
+				// Work around it for this specific class (fingers-crossed) by
+				// referencing the protected enum via its subclass name
+				assignStmts = strings.Replace(assignStmts, c.ClassName+`::`, cppSubclassName(c)+`::`, -1)
+
+				// Also need to scan parent classes (e.g. QColumnView friend
+				// functions refer to its parent QAbstractItemView::State)
+				for _, classInherit := range c.AllInheritsClassInfo() {
+					assignStmts = strings.Replace(assignStmts, classInherit.Class.ClassName+`::`, cppSubclassName(c)+`::`, -1)
+				}
+
+				//
+
+				ret.WriteString(
+					m.ReturnType.RenderTypeCabi() + " " + cabiProtectedBaseName(c, m) + "(bool* _dynamic_cast_ok, " + emitParametersCabi(m, ifv(m.IsConst, "const ", "")+"void*") + ") {\n" +
```

Comment:
The safety check with dynamic_cast is not really needed here - when using qualified lookup, the call is done using static dispatch meaning that it does not actually touch the vbtable/derived type/anything like that.